CosmicReco/src/CosmicTrackFinder_module.cc

```diff
@@ -186,13 +191,16 @@ namespace mu2e{
    const Tracker* tracker = th.get();
   
    _stResult.run = &run;
-   auto const& srep = _strawResponse_h.get(run.id());
    _tfit.setTracker  (tracker);
-   _tfit.setStrawResponse (srep);
+   
+
    }
 
   void CosmicTrackFinder::produce(art::Event& event ) {
-     
+
+     auto _srep = _strawResponse_h.getPtr(event.id());
```

Comment:
To avoid shared_ptr overhead mentioned by Rob, you could use a reference instead by adding a line:
`StrawResponse const& srep = * _srep.get(); // (i think this is the correct way)`

which dereferences the shared pointer and stores the reference to the object in `srep`, which you then pass around instead of `_srep`.

I think Rob mentioned preferring this method to avoid the overhead caused by passing around / creating many copies of `shared_ptr` / `StrawResponse::cptr`. Doing it this way would then mean changing

`StrawResponse::cptr` types go to `StrawResponse const&`

Reply:
also, on my list. Just waiting for Rob to clarify regarding if the const Tracker& can stay for now. All this code is just stepping stones to the end result, there will be many more changes as this becomes more advanced. For now my focus is getting BTrk ready and then improving the performance of this code will follow